cmd/fs-v1.go

```diff
@@ -632,8 +636,9 @@ func (fs fsObjects) PutObject(bucket string, object string, data *hash.Reader, m
 		errorIf(err, "Failed to create object %s/%s", bucket, object)
 		return ObjectInfo{}, toObjectErr(err, bucket, object)
 	}
-
-	metadata["etag"] = hex.EncodeToString(data.MD5Current())
+	if metadata["etag"] == "" {
```

Comment:
This is not needed, "etag" will always be the MD5Current()

Reply:
for large uploads, the ETag generated during multipart upload to backend will not match the ETag generated using single Put operation for caching to FS. Hence the cache Put operation needs to pass the Etag in the metadata headers so that the backend's ETag can be used on the cached object as well. This check is required unless caching also uses multi-part put operation.

Reply:
This looks like design bug. It's changing code to cater  for caching requirement .. such type of spillage shouldn't happen.